etna/whatson/models.py

```diff
@@ -289,6 +290,20 @@ def filter_form_data(self, filter_form_cleaned_data):
 
         return events
 
+    def serve(self, request):
+        # Check if the request comes from JavaScript
+        # 'JS-Request' is a custom header added by the frontend
+        # If so, return just the event listing, not the full page
+        if request.headers.get("JS-Request"):
+            print(self.get_context(request))
+
+            return render(
+                request, "includes/whats-on-listing.html", self.get_context(request)
+            )
+        else:
+            # Display whats on page as usual
+            return super(WhatsOnPage, self).serve(request)
```

Comment:
```suggestion
            return super().serve(request)
```

You should be fine to use the short form here.